### ISSUE TITLE
Problem using GUEST App to share with external colleague. (oc server 10.0.10.4)

### ISSUE THREAD

**Issue by malvivent7** [owncloud/core#34567]
### Steps to reproduce
1.Installed GUEST app 0.70
2. Created a directory to share with guest
3. Shared with guest account (throug inserting an email)

### Expected behaviour
Guest user correctly login in owncloud (throug web and throug desktop client)and i expect to see the shared folder and its contents (the user is in form of <name>.<surname>@gmail.com in this situation but happens with any email inserted as guest user)
### Actual behaviour
Nothing in the guest users's folder and in the log i can see a lot of this entries Storage wrapper 'oc_readonly' was not registered via the 'OC_Filesystem - preSetup' hook which could cause potential problems.

### Server configuration
**Operating system**:

**Web server:**

**Database:**

**PHP version:**

**ownCloud version:** (see ownCloud admin page)

**Updated from an older ownCloud or fresh install:**

**Where did you install ownCloud from:**

**Signing status (ownCloud 9.0 and above):**

```
Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results into https://gist.github.com/ and puth the link here.
```
No errors have been found.
[config_report_20190220.txt](https://github.com/owncloud/core/files/2883574/config_report_20190220.txt)

**The content of config/config.php:**

```
Log in to the web-UI with an administrator account and click on
'admin' -> 'Generate Config Report' -> 'Download ownCloud config report'
This report includes the config.php settings, the list of activated apps
and other details in a well sanitized form.

or 

If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your ownCloud installation folder

*ATTENTION:* Do not post your config.php file in public as is. Please use one of the above
methods whenever possible. Both, the generated reports from the web-ui and from occ config:list
consistently remove sensitive data. You still may want to review the report before sending.
If done manually then it is critical for your own privacy to dilligently
remove *all* host names, passwords, usernames, salts and other credentials before posting.
You should assume that attackers find such information and will use them against your systems.
```

**List of activated apps:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your ownCloud installation folder.
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
NO
**Are you using encryption:** yes/no
NO
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
LDAP
#### LDAP configuration (delete this part if not used)

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your ownCloud installation folder
+-------------------------------+------------------------------------------------------------------------------------------------+
| hasMemberOfFilterSupport      | 1                                                                                              |
| hasPagedResultSupport         |                                                                                                |
| homeFolderNamingRule          |                                                                                                |
| lastJpegPhotoLookup           | 0                                                                                              |
| ldapAgentName                 | CN=yyyyyy,OU=yyyyy,DC=yyyyy,DC=it                                                     |
| ldapAgentPassword             | ***                                                                                            |
| ldapAttributesForGroupSearch  |                                                                                                |
| ldapAttributesForUserSearch   |                                                                                                |
| ldapBackupHost                | ldap://xxx.xxx.xxx.xxx                                                                          |
| ldapBackupPort                | 389                                                                                            |
| ldapBase                      | OU=yyyyy,DC=yyyy,DC=it                                                                       |
| ldapBaseGroups                | ou=yyyy,dc=yyyyy,dc=it                                                                       |
| ldapBaseUsers                 | ou=yyyyy,dc=yyyyy,dc=it                                                                       |
| ldapCacheTTL                  | 60                                                                                             |
| ldapConfigurationActive       | 1                                                                                              |
| ldapDynamicGroupMemberURL     |                                                                                                |
| ldapEmailAttribute            | mail                                                                                           |
| ldapExperiencedAdmin          | 0                                                                                              |
| ldapExpertUUIDGroupAttr       |                                                                                                |
| ldapExpertUUIDUserAttr        | displayname                                                                                    |
| ldapExpertUsernameAttr        |                                                                                                |
| ldapGroupDisplayName          | name                                                                                           |
| ldapGroupFilter               | sAMAccountType=268435456                                                                       |
| ldapGroupFilterGroups         |                                                                                                |
| ldapGroupFilterMode           | 1                                                                                              |
| ldapGroupFilterObjectclass    |                                                                                                |
| ldapGroupMemberAssocAttr      | member                                                                                         |
| ldapHost                      | ldap://xxx.xxx.xxx.xxx                                                                          |
| ldapIgnoreNamingRules         |                                                                                                |
| ldapLoginFilter               | (&(&(|(objectclass=organizationalPerson)))(|(samaccountname=%uid)(|(distinguishedName=%uid)))) |
| ldapLoginFilterAttributes     | distinguishedName                                                                              |
| ldapLoginFilterEmail          | 0                                                                                              |
| ldapLoginFilterMode           | 1                                                                                              |
| ldapLoginFilterUsername       | 1                                                                                              |
| ldapNestedGroups              | 1                                                                                              |
| ldapOverrideMainServer        | 0                                                                                              |
| ldapPagingSize                | 500                                                                                            |
| ldapPort                      | 389                                                                                            |
| ldapQuotaAttribute            |                                                                                                |
| ldapQuotaDefault              |                                                                                                |
| ldapTLS                       | 0                                                                                              |
| ldapUserDisplayName           | displayname                                                                                    |
| ldapUserDisplayName2          |                                                                                                |
| ldapUserFilter                | (&(&(|(objectclass=organizationalPerson)))(userAccountControl=512))                            |
| ldapUserFilterGroups          |                                                                                                |
| ldapUserFilterMode            | 1                                                                                              |
| ldapUserFilterObjectclass     | organizationalPerson                                                                           |
| ldapUserName                  | samaccountname                                                                                 |
| ldapUuidGroupAttribute        | auto                                                                                           |
| ldapUuidUserAttribute         | auto                                                                                           |
| turnOffCertCheck              | 0                                                                                              |
| useMemberOfToDetectMembership | 1                                                                                              |
+-------------------------------+------------------------------------------------------------------------------------------------+
Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```

### Client configuration
**Browser:**
Any browser
**Operating system:**
win 10 1809 build 17763.292
#### Web server error log
```
Insert your webserver log here
```

#### ownCloud log (data/owncloud.log)
```
Insert your ownCloud log here
```

#### Browser log
```
Insert your browser log here, this could for example include:

[owncloud.zip](https://github.com/owncloud/core/files/2883638/owncloud.zip)

a) The javascript console log
b) The network log 
c) ...
```

**Comment by ownclouders** [comment]
[GitMate.io](https://gitmate.io) thinks the contributor most likely able to help you is @ownclouders.

Possibly related issues are https://github.com/owncloud/core/issues/30063 (OC 10.0.4 - Error while trying to create shared mount), https://github.com/owncloud/core/issues/30098 (OC 10.0.4.4 - Issue with Installing and Managing Apps), https://github.com/owncloud/core/issues/30094 (Fresh install of 10.0.4 fails with apps issue on shared server with no sudo), https://github.com/owncloud/core/issues/28265 (Hi,), and https://github.com/owncloud/core/issues/29158 (Dear Support, ).

**Comment by davitol** [comment]
Tested In oC 10.0.10 and Guests 0.7.0 and works for me.

### Test steps

1. admin user creates a folder with files
2. admin user shares the folder creating a Guest user
3. Guest user logs in his account
4. Guest user is able to browse the files of the folder


So it works fine in my Server

@malvivent7 Can you check if the path where the Guest folder root is created in your backend has write permissions? Feel free to provide any feedback you can consider worthy. Thanks for reporting!

**Comment by malvivent7** [comment]
Hi davitool, i have created folder and shared with guest through e-mail (correctly received see in image)
![image](https://user-images.githubusercontent.com/33978878/53082969-47a76100-34fe-11e9-91f1-708ab1f29bc8.png)
After login of guest user (set a password and login) this is the content of home in owncloud
![image](https://user-images.githubusercontent.com/33978878/53083254-d1efc500-34fe-11e9-9bd5-8d2c70e5405e.png) and in log "\/owncloud\/index.php\/apps\/files\/ajax\/getstoragestats.php?dir=%2F","message":"Storage wrapper 'oc_readonly' was not registered via the 'OC_Filesystem - preSetup' hook which could cause potential problems."
This is the root of my guest user 
![image](https://user-images.githubusercontent.com/33978878/53083785-d49eea00-34ff-11e9-8d59-1a037fa40af1.png) and here (like other position under /opt/ocdata) www-data has full control

**Comment by malvivent7** [comment]
Problem could be related that username has '@'?

**Comment by davitol** [comment]
> Problem could be related that username has '@'?

@malvivent7  username contains an '@' as:

```usern@me@gmail.com``` or ```username@gmail.com```

If it is like the second option, it works for me

**Comment by malvivent7** [comment]
It's like second example user.name@gmail.com

**Comment by davitol** [comment]
Then the origing of your issue is not probably related to the '@'

**Comment by malvivent7** [comment]
if i login with desktop client (latest stable version 2.5.3 build 11250) with my guest user i dont see anything
![image](https://user-images.githubusercontent.com/33978878/53088074-fea8da00-3508-11e9-88ff-bd16fcf28b3e.png)
and this is the log of desktop client
![image](https://user-images.githubusercontent.com/33978878/53088159-3dd72b00-3509-11e9-96cb-11901d161415.png)

**Comment by davitol** [comment]
Tested with  Desktop Version 2.4.3 (build 10347) it also went fine for me.
Did yoy check or do you have access to check if the folder in server used as backend has write permissions?

**Comment by malvivent7** [comment]
yes i have access to the server. Through WebUI i can see that my guest user is configured in this manner (see the photo)
![image](https://user-images.githubusercontent.com/33978878/53097663-4d16a280-3522-11e9-8ce3-aacd81070486.png)
From SSH console the /opt/ocdata directory and sub-directory are owned by www-data user and group and rights are 0755
![image](https://user-images.githubusercontent.com/33978878/53098081-29a02780-3523-11e9-8532-ed4c88d08ebd.png)

**Comment by malvivent7** [comment]
Davitol from the email that my guest has received if i click on the link to visualize contents i can go in the appropriate webpage (see the screenshot)
![image](https://user-images.githubusercontent.com/33978878/53098706-87813f00-3524-11e9-9a6a-1c639dfaa0d0.png) but if i click on the home button 
![image](https://user-images.githubusercontent.com/33978878/53098768-aaabee80-3524-11e9-8986-01ce15a7612c.png)
i'm unable to return to the shared folder
![image](https://user-images.githubusercontent.com/33978878/53098818-cadbad80-3524-11e9-9182-8b74d9a02e9f.png)

**Comment by malvivent7** [comment]
![image](https://user-images.githubusercontent.com/33978878/53099015-2b6aea80-3525-11e9-93b6-da3664238ef9.png)

**Comment by malvivent7** [comment]
I have recreated guest user's sharing from scratch (also uninstalling and reinstalling guest apps) when i share in the owncloud log i can see this message:
{"reqId":"Z7O1rJ4YtPlyc97fufq0","level":3,"time":"2019-02-20T16:22:05+01:00","remoteAddr":"xxx.xxx.xxx.xxx","user":"YYYYY_yyyyy","app":"no app in context","method":"PUT","url":"\/owncloud\/apps\/guests\/users","message":"User backend OC\\User\\Database is returning home: \/opt\/ocdata\/xxxxx.xxxxxxx@gmail.com for user: xxxxx.xxxxxxxx@gmail.com which differs from existing value: "}

**Comment by davitol** [comment]
> I have recreated guest user's sharing from scratch (also uninstalling and reinstalling guest apps) when i share in the owncloud log i can see this message:
> {"reqId":"Z7O1rJ4YtPlyc97fufq0","level":3,"time":"2019-02-20T16:22:05+01:00","remoteAddr":"xxx.xxx.xxx.xxx","user":"YYYYY_yyyyy","app":"no app in context","method":"PUT","url":"/owncloud/apps/guests/users","message":"User backend OC\User\Database is returning home: /opt/ocdata/xxxxx.xxxxxxx@gmail.com for user: [xxxxx.xxxxxxxx@gmail.com](mailto:xxxxx.xxxxxxxx@gmail.com) which differs from existing value: "}

This log is unrelated to the issue. It was already reported here: https://github.com/owncloud/core/issues/30853

**Comment by malvivent7** [comment]
Mean time from package-manager (apt update and apt upgrade) i received the new version of owncloud; installed (10.1.0.4) and all goes smoothly but problem persist.

**Comment by davitol** [comment]
@malvivent7 The part about it is not possible to upload or create files/folders works as designed:
<img width="1113" alt="screen shot 2019-02-20 at 16 43 20" src="https://user-images.githubusercontent.com/10596560/53104743-38d9a200-3530-11e9-9610-3c9ecd20b7ce.png">




> 2. Created a directory to share with guest

The name of the directory shared in the step 2, is Shared as it is shown in the screenshot?

**Comment by malvivent7** [comment]
No it's hamdigest look in my screenshot
![image](https://user-images.githubusercontent.com/33978878/53105421-6541ee00-3531-11e9-9b2e-071174db0df1.png)
and this is the share detail
![image](https://user-images.githubusercontent.com/33978878/53105484-8276bc80-3531-11e9-9653-ef5a91dec80e.png)

**Comment by davitol** [comment]
> No it's hamdigest look in my screenshot

Then idk why it appears in the screenshot breadcrumbs Shared > hamdigest

 hamdigest folder should appear as 1st level in the guests user account

**Comment by malvivent7** [comment]
Sostantialy i have shared hamdigest that is in my root to guest user and misteriously in the root folder of the guest no hamdigest folder is seen (for example like your screenshot) but i dont know why; all passage are produced through webui

**Comment by malvivent7** [comment]
First created folder from user A;
than user A share folder to user B inserting email as guest
user B receive an email by witch he can authenticate to owncloud changeing password
user B than login 
user B dosen't see nothing in his root
but if user B click on the link provided by email to see shared contents he can see folder structured as "Shared->hamdigest"

**Comment by malvivent7** [comment]
i'm struggling with this problem because that app it's very helpful for our activity

**Comment by PVince81** [comment]
@malvivent7 you forgot to attach your config.php but I guess you have set the option "share_folder" and are experiencing this bug related to share folder and guests: https://github.com/owncloud/core/pull/34395

**Comment by malvivent7** [comment]
> @malvivent7 you forgot to attach your config.php but I guess you have set the option "share_folder" and are experiencing this bug related to share folder and guests: #34395

Thanks PVince81 for your reply i have attached opening this issue my config in txt format (you can retrive it on top of my first post) in my config.php share_folder is 
'share_folder' => '/Shared', (i think it's the default) but i experience same bug mentioned #34395

**Comment by malvivent7** [comment]
For your convenience i re-attach file config
[config_report_20190220.txt](https://github.com/owncloud/core/files/2888979/config_report_20190220.txt)

**Comment by PVince81** [comment]
ok, the bug in question will be fixed in the upcoming 10.2.0 release.

as a workaround you could remove the "share_folder" setting, but that might affect your existing users.

or  you could apply https://github.com/owncloud/core/pull/34395 as a patch (please do so on a test system first).

in any case, I think this ticket can be closed as it's resolved by the patch in question.
